### ISSUE TITLE
Add Sign in with Google info to Site Health

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#9571]
## Feature Description

The Client ID used for Sign in With Google should be surfaced in the Site Health UI, as well as any other settings (eg. sign in button styles, new user account creation, etc.).

The number of users who have used SiWG should be shown as at least an absolute number (and possibly a percentage), based on the usermeta values saved for each user who successfully signs in using SiwG.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When the `signInWithGoogleModule` is enabled and the Sign in with Google module is connected, the following fields should be shown in Site Kit's Site Health:
  * Sign in with Google Client ID
  * Sign in with Google Shape
  * Sign in with Google Text
  * Sign in with Google Theme
  * Sign in with Google One-tap Enabled
  * The absolute number of users who have authenticated using Sign in with Google
  * A "Tag placement" check, similar to other modules like Analytics, for the Sign in with Google JS tag. The check should be made against the WordPress login page and (if the module is present) the WooCommerce login page.

## Implementation Brief

* [x] Update `includes/Modules/Sign_In_With_Google/Settings.php` to include both the label and settings value for each supported field to match the js definitions of theses constants:
   * https://github.com/google/site-kit-wp/blob/cc981876c3b654437e965cb66f8a09c9d110dba1/includes/Modules/Sign_In_With_Google/Settings.php#L26-L36
   * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/assets/js/modules/sign-in-with-google/datastore/constants.js#L19-L64
   * Update any existing uses of these constants in the rest of the php SiwG implementation.

* [x] Update `includes/Core/Site_Health/Debug_Data.php`
   * Add a new method `get_sign_in_with_google_fields`:
      * Instantiate a new instance of `use Google\Site_Kit\Modules\Sign_In_With_Google\Settings` and get the module settings.
         * Return each module setting as a new debug field with the titles in the AC:
            * For the style settings use the `label` value updated above in `includes/Modules/Sign_In_With_Google/Settings.php`.
            * For the clientID, return the clientID as stored.
            * For the One-tap Enabled setting return the string `Enabled` or `Disabled` based on the boolean value in the database.
      * Get a count of the key `SIGN_IN_WITH_GOOGLE_USER_ID_OPTION ` in the usermeta table with the following query:
         * ```
             $wpdb->query(
             	$wpdb->prepare( "SELECT count(id) FROM $wpdb->usermeta WHERE meta_key = %s", Sign_In_With_Google_Settings::SIGN_IN_WITH_GOOGLE_USER_ID_OPTION )
             );
             ```
   * Update the existing `get_fields` method, calling the new `get_sign_in_with_google_fields` method and merge it with the `$fields` array:
      * Wrap the call in the `Feature_Flags::enabled( 'signInWithGoogleModule' )` call to only return the fields when the feature flag is enabled.
      * Create a new instance of the `Sign_In_With_Google` module and wrap the call in a check for `$sign_in_with_google->is_connected()` to only render the fields when the module is connected.

* [ ] Create a new file `includes/Core/Site_Health/Sign_In_Button_Placement.php`, [this will take a similar approach to some elements of the Tag_Placement class, but the render method and future requirements warrant a new class](https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php):
   * Create a `register` method which registers the new test as a direct check:
      * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L72-L75
   * Create a new check method `sign_in_button_test`:
      * Set up the result to be returned at the end of the check, updating the label to `Sign In Button Placement` and the test prop to `sign_in_button_test`:
         * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L102-L112
      * Create an empty `$description` array. 
      * Get the content of the current wp-login page using this technique, replacing `home_url` with `wp_login_url`:
         * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L132-L133
         * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L153
      * Search for the presence of the translated tag name using the following technique:
         * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Modules/Module_With_Tag_Trait.php#L36-L42
         * If the tag is found add `Sign in button added to the WordPress login page.` to the `$description`, otherwise add `Sign in button missing on the WordPress login page.`
      * Check if WooCommerce is active using the `is_active` method on the `WooCommerce` class, if it's active do the following:
         * Get the content of the current WooCommerce login page using this technique, replacing `home_url` with `wc_get_page_id( 'myaccount' )`:
            * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L132-L133
            * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Site_Health/Tag_Placement.php#L153
         * Search for the presence of the translated tag name using the following technique:
            * https://github.com/google/site-kit-wp/blob/dcaa4b3e42a15e361bf947f1c3bf213367572b56/includes/Core/Modules/Module_With_Tag_Trait.php#L36-L42
            * If the tag is found add `Sign in button added to WooCommerce login page.` to the `$description`, otherwise add `Sign in button missing on the WooCommerce login page.`
      * Join the `$description` array into a string by commas and update and return `$result`.

* [ ] Update `includes/Core/Site_Health/Site_Health.php`, import the new class `Sign_In_Button_Placement` to a new const.
   * Within the `register` method, call `$this->sign_in_with_google->register()` if `Feature_Flags::enabled( 'signInWithGoogleModule' )` is true and the module is connected using the module `is_connected` method.

### Test Coverage

* Update `tests/phpunit/integration/Core/Site_Health/Debug_DataTest.php` to confirm the fields are added as expected.
* Create new tests for the new `Sign_In_Button_Placement` check.

## QA Brief

* Go to Site health and check the status tab for the Tag Placement check and the Info tab for other Site Health data.
* Verify that the new Sign in with Google data appears when the module is connected and doesn't when it is not.
* When WooCommerce is installed and activated, the Tag Placement check for SiwG will be split into two lines as below. However, #9340 needs to be merged before the SiwG button will ever be displayed on the My Account Woo Commerce login page. If you simply enable WooCommerce, perform the basic setup and activate the plugin, the Site Health info will show that and that the Tag is not placed (understandably). The success case will have to be tested as part of #9340.

![Image](https://github.com/user-attachments/assets/46f9fe9f-5ac7-4f73-8da4-630afbc31ca4)


## Changelog entry

* Add Sign in with Google to Site Health.

**Comment by tofumatt** [comment]
IB ✅

**Comment by jimmymadon** [comment]
> ITEM 1

I've fixed this in a [follow up PR](https://github.com/google/site-kit-wp/pull/9676).

> ITEM 2

My apologies, I should have added more details to the QAB which I have done now. The WordPress login page script that you have identified is correct - Site Kite Site Health checks for the `<-- Sign in with Google button added by Site Kit -->` HTML comment. This is present when the SiwG module is set up and connected. However, as the QAB now suggests, #9340 adds the button on the WooCommerce page so this will have to be tested then.

**Comment by kelvinballoo** [comment]
### QA Update ⚠ 

One more item I realised and maybe that's not an issue at all also.
The AC lists the field '**The absolute number of users who have authenticated using Sign in with Google**' to be appeared under SK's site health.

The only discrepancy is that on the actual field, it's written '**Sign in with Google Number of users who have authenticated using Sign in with Google**'

Question is, should we change the sentence from '**Sign in with Google Number of users who have authenticated using Sign in with Google**' to '**The absolute number of users who have authenticated using Sign in with Google**' ?

Screenshot is attached for reference:
<Details>

![Image](https://github.com/user-attachments/assets/33ff75ab-32ac-4448-adf4-2fbc62a1cf10)

</Details>
 
________________________________________________________________________

**ITEM 1** ✅ 
This is verified good. It's now 'Rectangular' and not 'Rectagular'.
All the relevant items are also showing accordingly.

<Details>

- Sign in with Google Client ID
- Sign in with Google Shape
- Sign in with Google Text
- Sign in with Google Theme
- Sign in with Google One-tap Enabled
- The absolute number of users who have authenticated using Sign in with Google

![Image](https://github.com/user-attachments/assets/92caa31e-53b1-4802-be9b-499d671f21a7)

</Details>

_______________________________________________________________________________________

**ITEM 2** ✅ 
Verified tag placement not detected for WooCommerce.
This is good enough for this ticket.

WordPress login page also has the tag required in the page source.
<Details>
Tag Placement:

![Image](https://github.com/user-attachments/assets/6a2f0baa-64dd-4435-80c5-47d0f506e448)

WordPress login page:

![Image](https://github.com/user-attachments/assets/43c2f6d7-03e1-4e2b-bd66-0f020fcccf09)


</Details>

**Comment by jimmymadon** [comment]
> Question is, should we change the sentence from 'Sign in with Google Number of users who have authenticated using Sign in with Google' to 'The absolute number of users who have authenticated using Sign in with Google' ?

I actually did not like the "Sign in with Google" prefix to all of these fields and ONLY added this to maintain consistency with all other modules' Site Health fields. I think we should create a simple issue which adds a colon sign (:) after the module name for all of these fields for better clarity.

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

I agree on that @jimmymadon . 
Raised another ticket here : https://github.com/google/site-kit-wp/issues/9680

With that I'll, move this ticket to approval.


- All the relevant items are also showing accordingly.
   <Details>
   - Sign in with Google Client ID
   - Sign in with Google Shape
   - Sign in with Google Text
   - Sign in with Google Theme
   - Sign in with Google One-tap Enabled
   - The absolute number of users who have authenticated using Sign in with Google

   ![Image](https://github.com/user-attachments/assets/92caa31e-53b1-4802-be9b-499d671f21a7)

   </Details>

- Verified tag placement not detected for WooCommerce Login page and also ensured that for WordPress login page, it's detected.
   <Details>
   Tag Placement:

   ![Image](https://github.com/user-attachments/assets/6a2f0baa-64dd-4435-80c5-47d0f506e448)

   </Details>

- WordPress login page also has the tag required in the page source.
   <Details>
   WordPress login page:

   ![Image](https://github.com/user-attachments/assets/43c2f6d7-03e1-4e2b-bd66-0f020fcccf09)

   </Details>
